### DISCUSSION TITLE
✈️ Allow to run shortcodes in the Before and After fields of the Dynamic Tag in the Video widget's Link field (Migrated from #10945)

### DISCUSSION THREAD

**Discussion by CapriciousSage** [elementor/elementor#33370] [Feature Request]
## Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
I am creating a membership site that will have a video library, and I have created a custom template that shows a Vimeo video based on the value of a custom field. Unfortunately, the advanced -> before/after content is not firing on the page. This project cannot go live without this bug being fixed, as the videos (the main thing being restricted to members) are showing to everyone until this problem is solved.

REPLICATION EXAMPLE:
Working Example using Excerpt:
Text Field, Content -> Dynamic -> Post Excerpt. 
Advanced -> Before: [wcm_restrict] -> After: [wcm_restrict]
Result: Member user sees the excerpt, non member does not (as expected)

Bug Example Using Video Widget:
Video Field, Content -> Dynamic -> Post Custom Field -> Custom Key -> featured-video
Advanced -> Before: [wcm_restrict] -> After: [wcm_restrict]
Result: Member user sees the excerpt, non member ALSO sees video. Tested with sample (non shortcode) text in the before and after, then viewed source. It does not trigger/show on the site.

THE BUG: The before/after does not fire correctly on the 'video' element when populated dynamically.

**Describe the solution you'd like**
Feedback from Support: "At the moment (and in general), the Video widget does not render shortcodes entered in its Before, After, and Fallback fields in its Dynamic tag's Advanced settings."

Solution: Make the Video widget render shortcodes entered in its Before, After, and Fallback fields in its Dynamic tag's Advanced settings as with other elements (like text fields, etc).

**Describe alternatives you've considered**
For my first attempt at a work around, I tried creating a "shortcode" field before and after the video element and adding [wcm_restrict] and [/wcm_restrict] respectively, but this doesn't work.

The Support Team has recommended that as another possible workaround, I could try "creating a custom shortcode function code (either added to the child theme's "functions.php" file or to a code snippet management tool such as Code Snippets) that will retrieve the post custom field with the Vimeo URL and render the Vimeo Video accordingly." - not ideal, involves a bit of faffery, but I'll give it a shot.

**Additional context**
Being that rendering shortcode (or content in general) in the Before, After, and Fallback fields of an element's Dynamic tag's Advanced settings is the standard function across elements supporting the use of dynamic tags, it stands to reason that the feature should be implemented consistently across the board. The ideal long term solution would be that any other elements not correctly allowing content &/or shortcode rendering in these Before, After, and Fallback fields also be fixed... but that's tomorrow's problem. Today calls for a fix to the video field so I can get my project live :-)

**Comment by DoronYosha** [comment]
I have the same problem, Did you find any solutio??

**Comment by nicholaszein** [comment]
Hi there,

### ✈️ This Feature Request has been migrated from issue #10945

if you still think this request should be considered for our roadmap, please cast your **⬆️ Upvote** on the first comment.

Don't worry, your comments, reactions, and participation in the old issue will be accounted for when considering this Feature Request for our Roadmap.

Happy voting! 🙂
